Evaluation/eval_tools.py

```diff
@@ -218,7 +218,15 @@ def create_roc_curve(self, df):
 
 def create_roc_ratio(x1, y1, x2, y2):
     sp = interpolate.interp1d(x2, y2)
```

Comment:
It looks like this problem of values beyond the range can be solved with just minor changes on `scipy` side by basically [changing the arguments](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.interp1d.html#scipy.interpolate.interp1d) to `interp1d()` as follows:
```suggestion
    sp = interpolate.interp1d(x2, y2, bounds_error=False, fill_value='extrapolate')
```
After this it will extrapolate (linearly by default) the values also beyond the reference curve range so that the fix you suggested shouldn't be needed.

On the other note, I've just noticed that the ratio plot isn't currently plotted for non WP-like curves:
  https://github.com/cms-tau-pog/TauMLTools/blob/5298854f3fb9219b17af539b70e1697426a20bdf/Evaluation/plot_roc.py#L170

Could you please fix it so that it isn't plotted only for the discriminator which is the same as the reference? That is:
 ``` python
    elif discr_name==ref_discr_name and curve_type==ref_curve_type:
```